### DISCUSSION TITLE
Help adding new scanner

### DISCUSSION THREAD

**Discussion by prabathbr** [villekf/OMEGA#41] [Q&A]
Hi  villekf,

Thank you very much for this interesting software. I am trying to add a new scanner based on Biograph to use sinogram data (I only have a header file .s.hdr + sinogram .s file and no other information on the scanner at the moment) and I am wondering which files to be edited ? Is it only "Biograph_Vision_main.m" or are there any other files to be matched. I couldn't exactly find variable names to be matched "Biograph_Vision_main.m" from the .s.hdr file. Is there any sample .s.hdr file from a ""Biograph_Vision" so that I can try to match values based on that to variables.

Thanks a lot!

**Comment by villekf** [comment]
You only need to edit one of the main files, for example `Biograph_Vision_main.m` (or the one for mCT). There, you need to edit the geometry parameters of the scanner, the (final) image size and the size of the sinogram. My Vision data is in ptd-format and it's a bit harder to extract the header information from it, but I'll paste some relevant information from an mCT header below, you can compare it with the `Biograph_mCT_main.m`.

!GENERAL IMAGE DATA:=
%study date (yyyy:mm:dd):=2017:05:06
%study time (hh:mm:ss GMT+00:00):=06:48:29
isotope name:=O-15
isotope gamma halflife (sec):=122.24
isotope branching factor:=0.999003
radiopharmaceutical:=Oxygen-water
%tracer injection date (yyyy:mm:dd):=2017:05:06
%tracer injection time (hh:mm:ss GMT+00:00):=06:44:21
tracer activity at time of injection (Bq):=5e+008
relative time of tracer injection (sec):=147044676
injected volume (ml):=0.0
image data byte order:=LITTLEENDIAN
%patient orientation:=FFS
!PET data type:=emission
data format:=sinogram
%compression:=on
%compressor version:=1.1
number format:=signed integer
!number of bytes per pixel:=2
number of dimensions:=3
matrix axis label[1]:=sinogram projections
matrix axis label[2]:=sinogram views
matrix axis label[3]:=number of sinograms
matrix size[1]:=400
matrix size[2]:=168
matrix size[3]:=621
scale factor (mm/pixel) [1]:=2.005
scale factor (degree/pixel) [2]:=1.07143
scale factor (mm/pixel) [3]:=2.027
horizontal bed translation:=stepped
start horizontal bed position (mm):=-2473.51
end horizontal bed position (mm):=-2473.51
start vertical bed position (mm):=141
%axial compression:=11
%maximum ring difference:=49
number of rings:=55
%number of TOF time bins:=13
%TOF mashing factor:=1
%sinogram type:=step and shoot
%number of segments:=9
%segment table:={109,97,97,75,75,53,53,31,31}
%total number of sinograms:=621
number of energy windows:=1
%energy window lower level (keV) [1]:=435
%energy window upper level (keV) [1]:=650
gantry tilt angle (degrees):=0.0
%coincidence window width (ns):=4.0625
number of scan data types:=14
